### PR TITLE
Update declare-empty-slice tenet

### PR DESCRIPTION
**Pull request by leilaes**

### CODE REVIEW

File: tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml [hunk 1]
```diff
@@ -22,13 +24,13 @@ tenets:
...
 
-          over 
+          over
 
           ```go
           t := []string{}
           ```
 
-          The former declares a nil slice value, while the latter is non-nil but zero-length. They are functionally equivalent—their `len` and `cap` are both zero—but the nil slice is the preferred style.
+          The former declares a nil slice value, while the latter is non-nil but zero-length. They are functionally equivalent ^`^ttheir `len` and `cap` are both zero ^`^tbut the nil slice is the preferred style.
```
Comment (BlakeMScurr) [inline tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml]: What do the carats (`^`) do here?

File: tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml [hunk 2]
```diff
@@ -51,4 +53,4 @@ tenets:
           end_offset
           go.composite_lit:
             child_count == 1
-            go.array_type
+            go.array_type
```
Comment (BlakeMScurr) [inline tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml]: Probably better to have a newline at the end of file - but it's not a big deal either way.

File: tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml [hunk 3]
```diff
@@ -4,7 +4,9 @@ funcs:
     body: |
       function(sliceType) {
         sliceType = sliceType.split(' ').join('');
-        return  sliceType.substring(0, sliceType.indexOf("{"))
+        if (sliceType.substring(sliceType.length - 4, sliceType.length) == "{}{}" && sliceType.indexOf("interface") !== 0)
```
Comment (BlakeMScurr) [inline tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml]: Can you please add a comment explaining this exception? I couldn't quite understand it. Like, why `sliceType.length - 4`, for example.

Comment (BlakeMScurr) [reply tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml]: If it has to do with the fact the `{}` in `interface{}` is part of the type, the same thing applies for `struct{}`.

Comment (leilaes) [reply tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml]: I am checking for empty interface, can you please give me an example of struct and I will include that too

File: tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml [hunk 4]
```diff
@@ -4,7 +4,9 @@ funcs:
     body: |
       function(sliceType) {
         sliceType = sliceType.split(' ').join('');
-        return  sliceType.substring(0, sliceType.indexOf("{"))
+        if (sliceType.substring(sliceType.length - 4, sliceType.length) == "{}{}" && (sliceType.indexOf("interface") !== 0 || sliceType.indexOf("struct") !== 0)) // This check is for empty interface or struct
```
Comment (BlakeMScurr) [inline tenets/codelingo/code-review-comments/declare-empty-slice/codelingo.yaml]:
```suggestion
        // This check is for empty interface or struct, which include `{}` as part of their type definition
        if (sliceType.substring(sliceType.length - 4, sliceType.length) == "{}{}" && (sliceType.indexOf("interface") !== 0 || sliceType.indexOf("struct") !== 0))
```

Readability

**Pull request merged**